application/src/main/java/org/opentripplanner/routing/alternativelegs/AlternativeLegs.java

```diff
@@ -39,20 +39,25 @@
  */
 public class AlternativeLegs {
 
+  public enum SearchDirection {
+    NEXT,
+    PREVIOUS,
+  }
+
```

Comment:
My suggestion is

```suggestion
  public enum SearchTime {
    AFTER,
    BEFORE
  }
```

There is no "global" types for this, and I am not sure we want one. These are the related enums I found - all have a specific meaning in the context they exist - so none of these are reusable:
```
SearchDirection {FORWARD, REVERSE}                                   (Raptor)  - search direction in time AND space
SortOrder       {STREET_AND_ARRIVAL_TIME, STREET_AND_DEPARTURE_TIME} (model.plan)
PageType        {PREVIOUS_PAGE, NEXT_PAGE}                           (model.plan.paging.cursor)
ListSection     {HEAD, TAIL}                                         (framework.collection)

arriveBy : boolean (routing.api.request.RouteRequest)
```